### PR TITLE
Fix identifier contingency list deserialization

### PR DESCRIPTION
**Pull request by EtienneLt**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
no

**What kind of change does this PR introduce?**
bug fix

**What is the current behavior?**
identifier contingency list are not deserialized this is probably just a oversight

**What is the new behavior (if this is a feature change)?**
it fixes it

### CODE REVIEW

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/ContingencyListDeserializer.java [hunk 1]
```diff
@@ -28,29 +28,39 @@ public ContingencyList deserialize(JsonParser parser, DeserializationContext des
...
+                    }
+                    case "list" -> {
                         ListOfContingencyListsDeserializer listOfContingencyListsDeserializer = new ListOfContingencyListsDeserializer();
                         return listOfContingencyListsDeserializer.deserialize(parser, deserializationContext);
-                    default:
-                        throw new IllegalStateException("Unexpected field: " + parser.getCurrentName());
+                    }
+                    case "identifier" -> {
+                        IdentifierContingencyListDeserializer identifierContingencyListDeserializer = new IdentifierContingencyListDeserializer();
+                        return identifierContingencyListDeserializer.deserialize(parser, deserializationContext);
```
Comment (EtienneLt) [inline contingency/contingency-api/src/main/java/com/powsybl/contingency/json/ContingencyListDeserializer.java:61]: here is the main change

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/ContingencyListDeserializer.java (hunk 1 above)
Comment (olperr1) [reply contingency/contingency-api/src/main/java/com/powsybl/contingency/json/ContingencyListDeserializer.java:61]: Ok, I read your PR to quickly. The attribute was already serialized, but its deserialization was not managed.

Could you add a unit test serializing and deserializing a `ContingencyList` with identifiers?

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListSerializer.java [hunk 2]
```diff
@@ -25,9 +25,9 @@ public IdentifierContingencyListSerializer() {
     @Override
     public void serialize(IdentifierContingencyList identifierContingencyList, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException {
         jsonGenerator.writeStartObject();
-        jsonGenerator.writeStringField("version", IdentifierContingencyList.getVersion());
         jsonGenerator.writeStringField("type", identifierContingencyList.getType());
         jsonGenerator.writeStringField("name", identifierContingencyList.getName());
+        jsonGenerator.writeStringField("version", IdentifierContingencyList.getVersion());
```
Comment (flo-dup) [inline contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListSerializer.java:30]: shouldn't you put it before name, like all the other serializers?

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierDeserializer.java [hunk 3]
```diff
@@ -37,56 +37,42 @@ public NetworkElementIdentifier deserialize(JsonParser parser, DeserializationCo
         String voltageLevelId1 = null;
         String voltageLevelId2 = null;
         String contingencyId = null;
-        String version = (String) deserializationContext.getAttribute("version");
+        String version = (String) deserializationContext.getAttribute("identifierListVersion");
```
Comment (flo-dup) [inline contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierDeserializer.java:40]: You created a utility method `JsonUtil::getSourceVersion` for that, shouldn't you use it?

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListDeserializer.java [hunk 4]
```diff
@@ -34,34 +34,23 @@ public IdentifierContingencyList deserialize(JsonParser parser, DeserializationC
...
         while (parser.nextToken() != JsonToken.END_OBJECT) {
             switch (parser.getCurrentName()) {
-                case "version":
+                case "version" -> {
                     version = parser.nextTextValue();
-                    deserializationContext.setAttribute("version", version);
-                    break;
-
-                case "identifiableType":
+                    deserializationContext.setAttribute("identifierListVersion", version);
```
Comment (flo-dup) [inline contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListDeserializer.java:41]: Same comment, execpt with `JsonUtil::setSourceVersion`

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListDeserializer.java [hunk 5]
```diff
@@ -34,34 +34,23 @@ public IdentifierContingencyList deserialize(JsonParser parser, DeserializationC
...
-                    break;
-
-                case "type":
-                    parser.nextToken();
-                    break;
-
-                case "identifiers":
+                }
+                case "name" -> name = parser.nextTextValue();
+                case "type" -> parser.nextToken();
```
Comment (flo-dup) [inline contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListDeserializer.java:48]: It's quite strange to read the type but ignoring it. In most cases (if not all?), we parse the type before calling this deserialize method. We should at least check that the type is the one expected don't you think?

**Review by flo-dup — CHANGES_REQUESTED** [review]

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListSerializer.java (hunk 2 above)
Comment (EtienneLt) [reply contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListSerializer.java:30]: I needed it after attribut type but I can put it before name

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierDeserializer.java (hunk 3 above)
Comment (EtienneLt) [reply contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierDeserializer.java:40]: I did not remember I'll chack that

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListSerializer.java (hunk 2 above)
Comment (flo-dup) [reply contingency/contingency-api/src/main/java/com/powsybl/contingency/json/IdentifierContingencyListSerializer.java:30]: yes, it's better to put it at the same place for all

**Review by flo-dup — APPROVED** [review]

**Pull request merged**